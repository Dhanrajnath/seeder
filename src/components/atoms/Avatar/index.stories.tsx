import { Meta, StoryObj } from "@storybook/react";
import Avatar, { IAvatarProps } from ".";
import AvatarSvg from "../../../../public/assets/avatar.svg";

export default {
    title: 'Atoms/Avatar',
    component: Avatar,
} as Meta;

type AvatarType = StoryObj<IAvatarProps>;


export const ImageAvatar: AvatarType = {
    args: {
        isImage: true,
        variant: 'circular',
        alt: "Profile Pic",
        src: AvatarSvg
    }
};

export const NonImageChildrenAvatar: AvatarType = {
    args: {
        isImage: false,
        variant: 'circular',
        children: 'AB'
    }
};

export const defaultAvatar: AvatarType = {
    args: {}
}