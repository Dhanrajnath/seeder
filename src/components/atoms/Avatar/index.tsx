import { Avatar as MUIAvatar, AvatarProps, styled } from '@mui/material';
import theme from '../../../utils/theme';

export interface IAvatarProps extends AvatarProps {
    children?: string;
    isImage: boolean;
    size?: number;
};

const StyledAvatar = styled(MUIAvatar)({
    height: theme.spacing(8),
    weight: theme.spacing(8),
    borderRadius: theme.spacing(3),
    gap: theme.spacing(2),
    fontWeight: 600,
    letterSpacing: 0.8
});

const Avatar = ({ children, isImage, size, ...props }: IAvatarProps) => {
    if (isImage) {
        return (
            <StyledAvatar {...props} />
        );
    } else {
        if (children) {
            return (
                <StyledAvatar
                    children={children}
                />
            );
        }
        return (
            <MUIAvatar {...props} />
        );
    }
};

export default Avatar;