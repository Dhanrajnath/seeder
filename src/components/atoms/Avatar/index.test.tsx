import { render } from "@testing-library/react";
import Avatar from ".";
import AvatarSvg from "../../../../public/assets/avatar.svg";

describe('Avatar tests', () => {
    it('renders Avatar with default props and should match the snapshot', () => {
        const { container } = render(<Avatar isImage={false} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders Avatar with image as prop and should match the snapshot', () => {
        const { container } = render(<Avatar isImage={true} variant="circular" alt="Profile Pic" src={AvatarSvg} />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders Avatar without image and with children', () => {
        const { container } = render(<Avatar isImage={false} variant="circular" children="ABC" />);

        expect(container.firstChild).toMatchSnapshot();
    })
});