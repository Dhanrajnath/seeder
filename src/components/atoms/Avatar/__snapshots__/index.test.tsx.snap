// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar tests renders Avatar with default props and should match the snapshot 1`] = `
<div
  class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-17o22dy-MuiAvatar-root"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
    data-testid="PersonIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
  </svg>
</div>
`;

exports[`Avatar tests renders Avatar with image as prop and should match the snapshot 1`] = `
<div
  class="MuiAvatar-root MuiAvatar-circular css-zc2ms7-MuiAvatar-root"
>
  <img
    alt="Profile Pic"
    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
    src="avatar.svg"
  />
</div>
`;

exports[`Avatar tests renders Avatar without image and with children 1`] = `
<div
  class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1tp6bdw-MuiAvatar-root"
>
  ABC
</div>
`;
